File: start.py
import os
import subprocess

# Caminhos dos scripts
install_script = "./install.sh"
proot_binary = "./libraries/proot"
startup_script = "./start.sh"  # ajuste conforme seu script real

# Dar permissão de execução
os.chmod(install_script, 0o755)
os.chmod(proot_binary, 0o755)

# Configurar variável de ambiente
env = os.environ.copy()
env["STARTUPSCRIPT"] = startup_script

print("Permissões alteradas, iniciando o install.sh...")

# Executar o install.sh em tempo real
process = subprocess.Popen(
    ["bash", install_script],
    stdout=subprocess.PIPE,
    stderr=subprocess.STDOUT,
    env=env,
    text=True
)

# Mostrar saída em tempo real
for line in process.stdout:
    print(line, end="")

process.wait()
print(f"install.sh finalizado com código: {process.returncode}")

File: start.js
import { spawn } from "child_process";
import { chmodSync } from "fs";

// Caminhos dos scripts/binaries
const installScript = "./install.sh";
const prootBinary = "./libraries/proot";
const prootArm64Binary = "./libraries/prootarm64";

// Dar permissão de execução
chmodSync(installScript, 0o755);
chmodSync(prootBinary, 0o755);
chmodSync(prootArm64Binary, 0o755);

console.log("Permissões alteradas, iniciando o install.sh...");

// Executar o install.sh em tempo real
const child = spawn("bash", [installScript], {
  stdio: "inherit" // exibe stdout e stderr em tempo real
});

child.on("exit", (code) => {
  console.log(`install.sh finalizado com código: ${code}`);
});
